source/backend/entities/Partials/Agreement.cs

```diff
@@ -11,5 +11,28 @@ public partial class PimsAgreement : StandardIdentityBaseAppEntity<long>, IBaseA
         [NotMapped]
         public override long Internal_Id { get => this.AgreementId; set => this.AgreementId = value; }
         #endregion
+
+        public bool IsEqual(PimsAgreement other)
```

Comment:
In my experience isEqual methods generally handle being passed a null other value.

Reply:
That's a good point, however, this one is not a Comparable but rather a helper method. I can make it more standard too but I am not sure we gain much from it.